src/libaktualizr/storage/sqlstorage.cc

```diff
@@ -783,10 +783,10 @@ void SQLStorage::storeEcuSerials(const EcuSerials& serials) {
       }
     }
 
-    EcuSerials::const_iterator it;
-    for (it = serials.begin() + 1; it != serials.end(); it++) {
-      auto statement = db.prepareStatement<std::string, std::string>("INSERT INTO ecu_serials VALUES (?,?,0);",
-                                                                     it->first.ToString(), it->second.ToString());
+    for (auto it = serials.cbegin() + 1; it != serials.cend(); it++) {
+      auto statement = db.prepareStatement<int64_t, std::string, std::string>(
+          "INSERT INTO ecu_serials(id,serial,hardware_id) VALUES (?,?,?);", it - serials.cbegin(), it->first.ToString(),
+          it->second.ToString());
```

Comment:
So the ID is just purely for sequencing, and we just trust that what we receive from the server is the order we should maintain? Not a bad idea to finally make it explicit instead of assuming primary is always first.

Reply:
Yes, the id is for sequencing.

I don't believe the order comes from the server, but the local config: https://github.com/advancedtelematic/aktualizr/blob/a51704b6fac25ce10f1fe024a81a01411225226f/src/libaktualizr/primary/initializer.cc#L68.

Hope I haven't overlooked anything but the intent is that any two devices with the same configuration (i.e: with same sets of ecu, deducted from configuration files named the same) would have their ecus in the same order, so that we can safely match them with the same device error code if several ecus fail.

e.g: `ecu1:fail1|ecu2:fail2` and never `ecu2:fail2:ecu1:fail1`

Reply:
> Not a bad idea to finally make it explicit instead of assuming primary is always first.

The api still expects a list of serials that start with the primary.